### ISSUE TITLE
.schematics file not loading

### ISSUE THREAD

**Issue by Deemunn** [maruohon/litematica#265]
**NOTE**:
* Before reporting an issue, please make sure it has not already been fixed in a later version! Also make sure you have the latest version of **both Litematica and malilib!** There are occasionally changes that **will require** both mods to be updated at the same time, or something will break! The very latest development versions are available here: https://masa.dy.fi/mcmods/client_mods/ so you can quickly check if you have the latest builds. The versions on CurseForge are not usually the very latest builds, but important bugfixes will also be uploaded there.
* Please don't post simple questions like "How do I do X?" on this issue tracker. Such help requests should happen on Discord instead: https://discord.gg/VNVG3Wk (or if the previous link is broken: https://discord.gg/scicraft -> `#masas-mods`)
* Don't post issues for mod versions made for Minecraft snapshots on this tracker. The snapshot versions aren't "properly" supported or tested, they are provided as quick ports (when I have the time to make them) in a "might work, mostly" basis, and those issues should rather be handled on Discord as well.

**Minecraft version**
Which Minecraft version are you playing on?
Fabric 1.16.4 v0.6.1.51, Litematica v0.0.0-dev.20201103.184101, Malilib: malilib-fabric-1.16.4-0.10.0-dev.21+arne.2

**Description of the bug**
When selecting a schematic downloaded from https://www.minecraft-schematics.com/, it says it cant load. Watched some tutorials and i am not wiser. The file is in the proper schematics folder. I expected at least a notification with the exact problem

**Screenshots**
![2020-11-30_20 57 05](https://user-images.githubusercontent.com/75275762/100658033-bad63c00-334e-11eb-9c4b-696586034e3e.png)

![Screenshot (3)](https://user-images.githubusercontent.com/75275762/100658116-d93c3780-334e-11eb-8d25-24a9ae7d70c3.png)

**Additional context**

If someone can try to explain this problem i'd really appreciate it.

**Comment by maruohon** [comment]
Can you link the exact schematic? But I think it's very likely that the schematic is in the newer Sponge schematic format. If it was made in 1.13+ then that's almost certainly the case.

The Sponge schematic format is currently only supported in the 1.12.2 versions of Litematica. That support should come to the other versions in the next bigger updates, hopefully around the end of the year. Until then you can use World Edit to paste it to a temporary world, and then save it from that with Litematica.

**To place a schematic file (*.schem) with World Edit**:
- Get World Edit for Fabric from here: https://www.curseforge.com/minecraft/mc-mods/worldedit/files/
- Put the schematic file in the `.minecraft/config/worldedit/schematics/` directory (create the directories first, use lower case names like shown)
- In-game run the command `//schem list` to check that the file is seen by the mod
- Then do `//schem load filename` to load it to memory
- Then do `//paste` to paste it to your player's position, or you can mark the position to paste to first by left clicking with a Wooden Axe

**Comment by Deemunn** [comment]
https://www.minecraft-schematics.com/schematic/15525/ is the exact one, tried another one same problem, i'll try the world edit one.

**Comment by Deemunn** [comment]
World edit worked, only problem is the schematic so i can build it on perhaps a server or some place without acces to world edit. Whats your advice with that?

**Comment by maruohon** [comment]
Like I said, you can use World Edit to paste it to a temporary (flat) world, and then you can save it with Litematica from that so you can use it with Litematica wherever you want.

**Comment by Deemunn** [comment]
Ah thanks misunderstood it, thanks for the help

**Comment by PenguinGiraffey** [comment]
i am having a similar problem. when i put my schematic in the folder it does not show up in-game. i am using the same version

**Comment by johnadams420** [comment]
sorry if i am too late

but would it work if I changed the .schematica at the end to .litematica in order to use it?

thanks for any help

**Comment by maruohon** [comment]
@johnadams420 No, that doesn't change the fact that the files use a completely different data structure.

If the `*.schematic` is in the old MCEdit/Schematica format from 1.12.2 or before, then it should load in the current Litematica versions. But if it's in the Sponge schematic format (for example from World Edit in 1.13+), then you will currently need to use World Edit to place it to a temporary world, and save it from that with Litematica, until the direct support for that format comes to Litematica. See my earlier reply for more info on how to do that.

**Comment by fendermitch** [comment]
![image](https://user-images.githubusercontent.com/129136940/228116442-20368825-d8b7-439f-9516-06f1c2e61735.png)
 So I have been trying to /schem load the file but Whenever I do it it says its doesnt exist But if i do /schem list it does? Can anyone help me out

**Comment by maruohon** [comment]
@fendermitch What format schematic is that, where did you get it from? Does it not load with Litematica either, or do you just want to paste it with World Edit for other reasons? Also in that screenshot are you using the actual full file name to try to load it? The messages just say `Castle` does not exist, but the list command shows longer names.

**Comment by MineCraftPro150** [comment]
i have the same problem but idk how to add the directory could you help?
here ill link in a photo
![image](https://github.com/maruohon/litematica/assets/161097869/d7f6a818-40f3-4cd2-9475-be706cd0d094)

**Comment by sanderkoolen2009-sys** [comment]
I know it is almost 2 years later but I just came accross the same problem do you know how to fix it?

**Comment by maruohon** [comment]
@sanderkoolen2009-sys There are many different issues discussed in this thread, are you referring to the original report or which thing exactly? If the original report, then which exact mod version are you using? Some schematics are named with the wrong file name extension, if it says `.schematic`, it might actually be a Sponge schematic and would need to have the extension `.schem` instead. (There are also some cases where they are named "even more wrong" such as a Litematica schematic named as `.schematic` instead of `.litematic`...) The current "old code" mod versions won't try to read the file as any other format than what the file name extension implies, so a wrong file name extension will prevent it from getting loaded.

**Comment by TylerSky** [comment]
@maruohon well im sorry to bother you but im struggling now. So ive tried litematic, worldedit, amulet, and even renaming it. However no luck. Death Star Trench (Star Wars) Minecraft Map https://www.planetminecraft.com/project/death-star-trench-star-wars/ 

I need some guidance. 🥲

**Comment by maruohon** [comment]
@TylerSky That file is named wrong, it should have the `.litematic` file name extension. After you rename it as such, then it will load fine at least in MC 1.21.4 and later versions (it was created in 1.21.4).
